src/Resources/skeleton/serializer/Encoder.php.txt

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Serializer;
+
+use Symfony\Component\Serializer\Encoder\DecoderInterface;
+use Symfony\Component\Serializer\Encoder\EncoderInterface;
+
+class {{ encoder_class_name }} implements EncoderInterface, DecoderInterface
+{
+    const FORMAT = '{{ format }}';
+    
+    public function encode($data, $format, array $context = array())
+    {
+        // TODO: return your encoded data
+        return '';
+    }
+
+    public function supportsEncoding($format): bool
```

Comment:
`?string $format` should work (not tested it)

Reply:
@dunglas - unfortunately the problem remains, tested on PHP 7.1.6

```
Fatal error: Declaration of YamlEncoder::supportsEncoding(?string $format) must be compatible with EncoderInterface::supportsEncoding($format)
```

Reply:
adding it relies on PHP 7.2 new features.